src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorLanguageServerClient.ts

```diff
@@ -106,34 +106,24 @@ export class RazorLanguageServerClient implements vscode.Disposable {
 
         try {
             this.logger.logMessage('Starting Razor Language Server...');
-            const startDisposable = this.client.start();
-            this.startDisposable = vscode.Disposable.from(startDisposable, didChangeStateDisposable);
+            await this.client.start();
             this.logger.logMessage('Server started, waiting for client to be ready...');
-            // tslint:disable-next-line: no-floating-promises
-            this.client.onReady().then(async () => {
-                if (currentState !== State.Running) {
-                    // Unexpected scenario, if we fall into this scenario the above onDidChangeState
-                    // handling will kill the start promise if we reach a certain retry threshold.
-                    return;
```

Comment:
I don't know if this is a worry, but its interesting that we used to return here if the server failed to start, but now we just let the error bubble up.. hopefully something deals with that

Reply:
According to [vscode documentation](https://github.com/Microsoft/vscode-languageserver-node/blob/main/README.md#3170-protocol-800-json-rpc-800-client-and-800-server), `onReady()` was replaced by `await this.client.start()`, so I _think_ it should be guaranteed that the server is a running state.